SharedSource/Development/UnitTesting/ThreadRunner.cs

```diff
@@ -56,26 +57,29 @@ public void Run ()
       Exception? lastException = null;
 
       // Use anonymous delegate to catch and store exceptions from the thread in the current scope.
-      Thread otherThread =
-        new Thread((ThreadStart)delegate
-          {
-            try
-            {
-              _threadStart();
-            }
-            catch (ThreadAbortException)
-            {
-              // Explicitely reset the ThreadAbortException
-              Thread.ResetAbort();
-              // Do not report exception in lastException, since aborting is expected behavior.
-            }
-            catch (Exception e)
-            {
-              lastException = e;
-            }
-          }
-         );
+      var threadStart = new ThreadStart(() =>
+      {
+        try
+        {
+          _threadStart();
+        }
+        catch (ThreadAbortException)
+        {
+          // Explicitely reset the ThreadAbortException
+          Thread.ResetAbort();
+          // Do not report exception in lastException, since aborting is expected behavior.
+        }
+        catch (Exception e)
+        {
+          lastException = e;
+        }
+      });
 
+      // TODO RM-8354: Remove ThreadRunner from SharedSource to allow reference to Core.Core
```

Comment:
Actually, the idea was to manually add a call to the boundary in the usage sites and then update those sites once RM-8354 is done.

Reply:
I know, I chose to do it this way because it was easier. Some references of `ThreadRunner` have no references to `Core` so they would not be changed, while other need to be changed. Because we already plan to change how this class works in the future this seemed like the easiest choice. Also it is just unit test code

Reply:
Am sorry, am not convinced. 

"just unit test code" isn't a good argument. While it's true that there's some tweaks and tricks you need to do when writing a unit test in an imperfect world (e.g. doing a private call), those are done because the API of the SUT takes precedence over the unit test and not everything can easily be refactored to be both sensibly constrained and have perfect testability.

The other thing is that this hides the dependency and the effect in a really horrible way.

=> I've looked at the other repositories and it looks like we don't have any external dependencies. If you do want to keep it here, please move ThreadRunner now (i.e. in this branch or - better - a separate branch that's happening before this one)